### ISSUE TITLE
Sort bookmarks by name

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#1066]
Currently, sorting by name (ascending and descending) is not available.

**Comment by lromanov** [comment]
It would be great if user could not only sort bookmark lists by name but move them manually in custom order as well.

**Comment by anishmittal2020** [comment]
Yes, also it would be nice if User has a search bar at the top to search for bookmark

**Comment by biodranik** [comment]
Even better to search bookmarks from the standard search bar.

**Comment by biodranik** [comment]
Another request is to allow custom sorting

**Comment by ioctl-user** [comment]
One more request -- add optional bookmarks directories or tags for users, who have a lot of lists that can be categorized.

**Comment by biodranik** [comment]
@ioctl-user please collect all related comments and issues about tags/subdirectories/folders/subfolders for bookmarks into a separate issue.

**Comment by ioctl-user** [comment]
Splitted to 2 new requests.

**Comment by metal450** [comment]
Is there any way to reorder bookmarks currently? Or you have to export them all to kml and then try to import them in the inverse-order of how you want them to appear...?

**Comment by biodranik** [comment]
Bookmarks should be ordered in the "last modified" order. There are some bugs though...

**Comment by metal450** [comment]
Thanks for the quick reply - and apologies, I realized I was a bit unclear. I was actually referring to the order of the bookmark folders/groups (rather than the order of the bookmarks themselves within the groups). I tried adding/editing/removing a bookmark from one of the "lower-down" groups, but that group remained where it was. The order of the groups appears random - i.e. it's not alphabetical, or any other order that makes them really easy to locate.

For example, I have a group for each place that I periodically travel to: Japan, Thailand, New York - plus some others, i.e. "Added On Phone," etc. The order they appear in seems to be random - they're all mixed up, so the only way to find the right one is to actually scroll through the entire list manually and look at each. I can't find any way, even by performing edits, to get them out of that order & into one that follows some logic - i.e. alphabetical, etc.

Thanks again

**Comment by Drkmstr310** [comment]
Is this anywhere on the roadmap? Looks like it's been hanging around for >2yrs - I also struggle with spending time scrolling back and forth trying to manually locate a bookmark list, as they seem to be in a randomized order. Even if custom sorting is more challenging, at least ordering by name would make it usable in the interm, as we could just prefix with a number to get them in the order we need.

**Comment by biodranik** [comment]
Yes, it's in the roadmap. The bug is tricky and requires some effort to investigate and fix (or workaround). Any volunteers are welcome.

**Comment by biodranik** [comment]
Bookmarks should be sorted. But there are issues to finish:

* Do not group tracks and bookmarks into separate sections but mix everything by name in one section
* Highlight which sorting is active now
* Make sure the sorting is correct with non-ASCII characters (depends on locale), e.g. use collation sorting
* Understand what's the difference in sorting "By date" and "By default" and maybe simplify it

**Comment by metal450** [comment]
Just to have the fixed sorting be by name would be an immense improvement over the current seemingly random order, in the interim. I realize this isn't the "final" solution, but even as a stepwise solution it would be infinitely better than delaying the "good" until it can be "perfect." Currently, with long bookmarks lists, there's just no way to find anything. So i.e. in the interim:

* Tracks & bookmarks can remain separate, and be sorted by name within their own sections (i.e. tracks by name in their section, bookmarks by name in theirs). These could be merged in one section later, it shouldn't be a blocker.
* No highlight necessary if sort order is fixed
* This seems easy (just a different function call?), but if not, again as a stepwise solution could just sort the ones that begin with an ASCII character for now, & wholly non-ASCII could be in their own section with the current ordering. Handling for other characters could be added gradually, and in the interim, users could simply prefix bookmark names with i.e. numbers to get a sensible order.
* Not applicable if only sorting by name

Again, not the final solution. But I don't think every related issue has to be 100% ideal in order to get to something a bit more usable.

**Comment by oleg-rswll** [comment]
Related to #5680
